src/Avalonia.Input/TappedGestureEventArgs.cs

```diff
@@ -0,0 +1,44 @@
+using Avalonia.Interactivity;
+using Avalonia.VisualTree;
+
+namespace Avalonia.Input
+{
+    public class DoubleTappedEventArgs : RoutedEventArgs
```

Comment:
Why do we have 3 separate classes instead of one?

Reply:
No particular reason, just using nearly same API with different classes as in UWP.
As for me it would be also fine to have single TappedEventArgs.

Reply:
It seems Microsoft has the mentality of code duplication in a lot of places in UWP. I've had similar discussions in the WinUI repo for new controls where they were making brand-new enums and similar properties instead of re-using or tweaking existing concepts.

Aside from not thinking things through enough to reach a good level of code-reuse, I think they do this to ensure they aren't locked in if they need to change something for one control but not another. It's a direct result of Microsoft's policy of no breaking changes. Personally, I miss the WPF mindset where they unified everything possible and you could see evidence of that throughout the entire framework. Everything was more logical and easier to remember.

Long story short, I am a fan of not repeating Microsoft's mistakes and searching for the best possible API design is quite important. Of course it must be balanced with WinUI code compatiblity. In this case we might be able to have both, at least from a framework standpoint.

What about making the common `TappedEventArgs` the base class for the other two?

```csharp
public class TappedEventArgs : RoutedEventArgs
public class DoubleTappedEventArgs : TappedEventArgs 
public class RightTappedEventArgs  : TappedEventArgs
```

Reply:
> Of course it must be balanced with WinUI code compatiblity

From that side we already is off from that, because in WinUI this events called DoubleTappedRoutedEventArgs / TappedRoutedEventArgs. See @grokys comment on previous PR for this API - https://github.com/AvaloniaUI/Avalonia/pull/3763#issuecomment-611441473

I will update this PR to use single TappedEventArgs for now. And I have doubts that we will need something specific for Double/Right tapped event args in the future.